app/src/main/java/net/gaast/giggity/Schedule.java

```diff
@@ -421,6 +424,170 @@ private void loadIcal(BufferedReader in) {
+				icon = conference.getString("logo");
+			}
+
+			links = new LinkedList<>();
+
+			//Adding website separately because it is not in social_links
+			if (conference.has("event_url")) {
+				Schedule.Link slink = new Link(conference.getString("event_url"), "Website");
+				links.addLast(slink);
+			}
+
+			//Adding ticket url separately because it is not in social_links
+			if (conference.has("ticket_url")) {
+				Schedule.Link slink = new Link(conference.getString("ticket_url"), "Ticket URL");
+				links.addLast(slink);
+			}
+
+			//Using social links of the event like facebook, google+, etc.
+			if (conference.has("social_links")) {
+				JSONArray linklist = conference.getJSONArray("social_links");
+				for (i = 0; i < linklist.length(); ++i) {
+					JSONObject link = linklist.getJSONObject(i);
+					Schedule.Link slink = new Link(link.getString("link"), link.getString("name"));
+					slink.setType(link.optString("type", null));
+					links.addLast(slink);
+				}
+			}
+
+
+			if (conference.has("microlocations")) {
+
+                /*Changing the flag after checking the organizer is using with microlocations
+				options enabled*/
+
+				hasMicrolocs = true;
+
+				//Getting microlocations to add latitude and longitude
+				JSONArray microlocations = conference.getJSONArray("microlocations");
+
+				for (i = 0; i < microlocations.length(); i++) {
+
+					JSONObject room = microlocations.getJSONObject(i);
+					locs.put(room.getString("name"), room.getString("longitude") + "," + room.getString("latitude"));
+				}
+			}
+
+
+			//The sessions are contained in the array present in an object
+			JSONArray events = conference.getJSONArray("sessions");
+
+			for (i = 0; i < events.length(); i++) {
+
+				JSONObject event = events.getJSONObject(i);
+				String uid = event.getString("id");
+				String title = event.getString("title");
+
+                /*Our date format is different and I changed getTimeInMillis() a bit to ignore "+08"
+				in second part to avoid error in integer parsing*/
+				String startTimeS = event.getString("start_time");
+				String endTimeS = event.getString("end_time");
+				Date startTime, endTime;
+				startTime = new Date(getTimeInMillis(startTimeS));
+				endTime = new Date(getTimeInMillis(endTimeS));
+				Schedule.Item item = new Schedule.Item(uid, title, startTime, endTime);
+				item.setDescription(event.getString("long_abstract"));
+
+				if (event.getString("signup_url") != "null") {
+					item.addLink(new Link(event.getString("signup_url")));
+				}
+
+
+				JSONObject microlocation = event.getJSONObject("microlocation");
+				String location = microlocation.getString("name");
+
+				if ((line = tentMap.get(location)) == null) {
+					line = new Schedule.Line(location, location);
+					tents.add(line);
+					tentMap.put(location, line);
+				}
+
+
+				//Getting value (latitude and longitude) from the map by key (name)
+
+				if (hasMicrolocs && line.getTitle()!=null && !line.getTitle().equals("")) {
```

Comment:
The one I was more concerned about it the one below.

But anyway, it's all not *that* important if you ensure that the input you provide for your conference is valid. Time's going to run out in a bit and we should allow for some time for people to upgrade to the new Giggity .apk. So I guess we should merge this.